### DISCUSSION TITLE
Teampass 3 API functionality?

### DISCUSSION THREAD

**Discussion by acartro** [nilsteampassnet/TeamPass#3756] [Q&A]
Hello, Teampass team.

I'm currently working on an integration with Teampass using the API, and I've noticed in the version 3 documentation it is mentioned that the API is still in development.

I wanted to ask:

1. Is the API for version 3 currently functional? If not, what is the estimated timeline for its full release?
2. Is the version 3 API completely different from the previous version, or are there similarities? Would I need to completely refactor my code or could some parts be reused?

I would appreciate any information you can provide to assist in my development process.

Thank you for your time and attention.

**Comment by nilsteampassnet** [comment]
Hello @acartro 

API is indeed in development and released features are functional.
The complete list is in [API usage part](https://documentation.teampass.net/#/api-basic?id=api-usage) of the documentation.

The way it works is more secure and relies now on jwt token. So you would have this part to adapt.
Regarding api functions, please have a look to the documentation but most of them have changed regarding V2.

Thank you

**Comment by nilsteampassnet** [comment]
No answer from author

**Comment by nosrio** [comment]
Hi! Do you have any updates on API functionality? I would like to automate items creation and files attachment to those credentials but the API do not seems to work.

**Reply by nilsteampassnet** [reply]
@nosrio 
No update has been done API regarding features.
It will but I need to focus first on the main "trunck".
In order to update my todo, what you expect would be "item creation" in position 1 and then "file attachment", correct?

**Reply by nosrio** [reply]
Yes, those two endpoint in that order of priority would be great. 
Also I 've been testing the "get item" api but i am getting an 500 error when using the "filter by label"

Example:

```
Request: /api/index.php/item/get

Response:

{
    "error": "Item id, label or description is mandatory"
}[
    {
        "id": 1,
        "label": "test1",
        "description": "<p>test1<br></p>",
        "pwd": "test1",
        "url": "",
        "login": "",
        "email": "",
        "viewed_no": 2,
        "fa_icon": "",
        "inactif": 0,
        "perso": 0,
        "id_tree": 1,
        "folder_label": "TeamA",
        "path": ""
    }
]

------------------------------------
Request: /api/index.php/item/get?label="%test1%"&like=1

Resposne: 500 Internal Server Error
```
Let me know if you need further info

**Comment by macalderonv** [comment]
Hello, 
I am also working on an integration with Teampass using the API and I see that the documentation for version 3 API is still under development.
I wanted to ask:

Is the API currently functional for version 3? 
If yes, can you tell me how to launch API requests?
If not, what is the estimated timeline for a full release?
I would appreciate any information you can provide to help me in my development process.

Thanks for your time and attention.
